src/core/src/op/cum_sum.cpp

```diff
@@ -47,76 +81,30 @@ void op::v0::CumSum::validate_and_infer_types() {
-        return make_shared<op::v0::CumSum>(new_args.at(0), new_args.at(1), m_exclusive, m_reverse);
+        return std::make_shared<CumSum>(new_args.at(0), new_args.at(1), m_exclusive, m_reverse);
     else {
-        return make_shared<op::v0::CumSum>(new_args.at(0), m_exclusive, m_reverse);
+        return std::make_shared<CumSum>(new_args.at(0), m_exclusive, m_reverse);
     }
 }
 
-namespace {
-template <element::Type_t DATA_ET, element::Type_t AXIS_ET>
-bool evaluate_cum_sum(TensorVector& outputs, const TensorVector& inputs, const bool exclusive, const bool reverse) {
-    using data_t = fundamental_type_for<DATA_ET>;
-    using axis_t = fundamental_type_for<AXIS_ET>;
-    ov::reference::cumsum<data_t, axis_t>(inputs[0].data<data_t>(),
-                                          inputs[1].data<axis_t>(),
-                                          outputs[0].data<data_t>(),
-                                          inputs[0].get_shape(),
-                                          exclusive,
-                                          reverse);
-    return true;
-}
-
-#define CUM_SUM_TYPE_CASE(a, ...)                                          \
-    case element::Type_t::a: {                                             \
-        OV_OP_SCOPE(OV_PP_CAT4(evaluate_cum_sum, _, a, AXIS_ET));          \
-        return evaluate_cum_sum<element::Type_t::a, AXIS_ET>(__VA_ARGS__); \
-    }
-
-template <element::Type_t AXIS_ET>
-bool evaluate(TensorVector& outputs, const TensorVector& inputs, const bool exclusive, const bool reverse) {
-    switch (inputs[0].get_element_type()) {
-        CUM_SUM_TYPE_CASE(f32, outputs, inputs, exclusive, reverse);
-    default:
-        return false;
-    }
-}
-
-bool evaluate_cum_sum(TensorVector& outputs, const TensorVector& inputs, const bool exclusive, const bool reverse) {
-    auto rc = true;
-    switch (inputs[1].get_element_type()) {
-        NGRAPH_TYPE_CASE(evaluate_cum_sum, i32, outputs, inputs, exclusive, reverse);
-        NGRAPH_TYPE_CASE(evaluate_cum_sum, i64, outputs, inputs, exclusive, reverse);
-    default:
-        rc = false;
-        break;
-    }
-    return rc;
-}
-}  // namespace
-
-bool op::v0::CumSum::evaluate(TensorVector& outputs, const TensorVector& inputs) const {
+bool CumSum::evaluate(TensorVector& outputs, const TensorVector& inputs) const {
     OV_OP_SCOPE(v0_CumSum_evaluate);
-    OPENVINO_ASSERT(inputs.size() == 2,
-                    "Invalid size of inputs argument of evaluate method of CumSum operation. Provided: ",
-                    inputs.size(),
-                    ". Expected: 2");
-    OPENVINO_ASSERT(outputs.size() == 1,
-                    "Invalid size of outputs argument of evaluate method of CumSum operation. Provided: ",
-                    outputs.size(),
-                    ". Expected: 1");
-
-    return evaluate_cum_sum(outputs, inputs, is_exclusive(), is_reverse());
+    OPENVINO_ASSERT(outputs.size() == 1);
+    OPENVINO_ASSERT(inputs.size() == 2);
```

Comment:
do we need these exceptions at all?
I suppose evaluate is called on already validated node

Reply:
As evaluate is public API, so at least number of inputs/outputs are checked and exception is thrown instead `segfault` when access tensors by `[]`